pkg/kubectl/cmd/logs.go

```diff
@@ -118,112 +134,111 @@ func NewCmdLogs(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.C
+	cmd.Flags().BoolVar(&o.Timestamps, "timestamps", o.Timestamps, "Include timestamps on each line in the log output")
+	cmd.Flags().Int64Var(&o.LimitBytes, "limit-bytes", o.LimitBytes, "Maximum bytes of logs to return. Defaults to no limit.")
+	cmd.Flags().BoolVarP(&o.Previous, "previous", "p", o.Previous, "If true, print the logs for the previous instance of the container in a pod if it exists.")
+	cmd.Flags().Int64Var(&o.Tail, "tail", o.Tail, "Lines of recent log file to display. Defaults to -1 with no selector, showing all log lines otherwise 10, if a selector is provided.")
+	cmd.Flags().StringVar(&o.SinceTime, "since-time", o.SinceTime, i18n.T("Only return logs after a specific date (RFC3339). Defaults to all logs. Only one of since-time / since may be used."))
+	cmd.Flags().DurationVar(&o.Since, "since", o.Since, "Only return logs newer than a relative duration like 5s, 2m, or 3h. Defaults to all logs. Only one of since-time / since may be used.")
+	cmd.Flags().StringVarP(&o.Container, "container", "c", o.Container, "Print the logs of this container")
+	cmd.Flags().BoolVar(&o.Interactive, "interactive", o.Interactive, "If true, prompt the user for input when required.")
 	cmd.Flags().MarkDeprecated("interactive", "This flag is no longer respected and there is no replacement.")
 	cmdutil.AddPodRunningTimeoutFlag(cmd, defaultPodLogsTimeout)
-	cmd.Flags().StringP("selector", "l", "", "Selector (label query) to filter on.")
+	cmd.Flags().StringVarP(&o.Selector, "selector", "l", o.Selector, "Selector (label query) to filter on.")
 	return cmd
 }
 
-func (o *LogsOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
-	containerName := cmdutil.GetFlagString(cmd, "container")
-	selector := cmdutil.GetFlagString(cmd, "selector")
+func (o *LogsOptions) calculateContainerName(args []string) error {
 	switch len(args) {
 	case 0:
-		if len(selector) == 0 {
-			return cmdutil.UsageErrorf(cmd, "%s", logsUsageStr)
+		if len(o.Selector) == 0 {
+			return fmt.Errorf(logsUsageStr)
 		}
 	case 1:
 		o.ResourceArg = args[0]
-		if len(selector) != 0 {
-			return cmdutil.UsageErrorf(cmd, "only a selector (-l) or a POD name is allowed")
+		if len(o.Selector) != 0 {
+			return fmt.Errorf("only a selector (-l) or a POD name is allowed")
 		}
 	case 2:
-		if cmd.Flag("container").Changed {
-			return cmdutil.UsageErrorf(cmd, "only one of -c or an inline [CONTAINER] arg is allowed")
+		if o.ContainerSpecified {
+			return fmt.Errorf("only one of -c or an inline [CONTAINER] arg is allowed")
 		}
 		o.ResourceArg = args[0]
-		containerName = args[1]
+		o.Container = args[1]
 	default:
-		return cmdutil.UsageErrorf(cmd, "%s", logsUsageStr)
-	}
-	var err error
-	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()
-	if err != nil {
-		return err
+		return fmt.Errorf("%s", logsUsageStr)
 	}
 
+	return nil
+}
+
+func (o *LogsOptions) calculatePodLogOptions() error {
 	logOptions := &api.PodLogOptions{
```

Comment:
@deads2k To switch this to corev1.PodLogOptions I'll also need to update / duplicate [validation.ValidatePodLogOptions](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/logs.go#L238)

Reply:
> @deads2k To switch this to corev1.PodLogOptions I'll also need to update / duplicate validation.ValidatePodLogOptions

That suggests we're doing clientside validation.  Just delete that code.